### ISSUE TITLE
Solidity keyword `using` not recognized => error: unrecognised token `using'

### ISSUE THREAD

**Issue by SvenMeyer** [hyperledger-labs/solang#592]
```solidity
pragma solidity ^0.7.2;

import "@openzeppelin/contracts/utils/EnumerableSet.sol"; // v3

contract OZimport {
    using EnumerableSet for EnumerableSet.AddressSet;
}
```

```
$ ./solang-linux -o build --importmap @openzeppelin=node_modules/@openzeppelin --target solana contracts_flat/OZimport.sol 

/home/sum/solang-test/contracts_flat/OZimport.sol:848:5-10: error: unrecognised token `using', expected "(", ";", "[", "abstract", "address", "bool", "bytes", "contract", "enum", "event", "false", "function", "import", "interface", "library", "mapping", "payable", "pragma", "string", "struct", "this", "true", Bytes, DocComment, Int, Uint, address, hexnumber, hexstring, identifier, number, rational, string
Line 848:
	using EnumerableSet for EnumerableSet.AddressSet;
	^^^^^
error: not all contracts are valid
```

**Comment by seanyoung** [comment]
I can't reproduce this issue.

The error looks like the parser grammar doesn't know about `using` in contract context which should work fine, that is where `using` should be.

In order to get my reproduction right, it would be useful to know the exact version of solang `solang --version` and where you got it from. Also where you running this (linux distribution or wsl or something else).

Thanks

**Comment by SvenMeyer** [comment]
fresh download today ...

```
$ ./solang-linux --version
solang version v0.1.9

$ uname -a
Linux xps15 5.4.159-1-MANJARO #1 SMP PREEMPT Fri Nov 12 20:25:49 UTC 2021 x86_64 GNU/Linux
```

**Comment by seanyoung** [comment]
> ```solidity
> pragma solidity ^0.7.2;
> 
> import "@openzeppelin/contracts/utils/EnumerableSet.sol"; // v3
> 
> contract OZimport {
>     using EnumerableSet for EnumerableSet.AddressSet;
> }
> ```
> 
> ```
> $ ./solang-linux -o build --importmap @openzeppelin=node_modules/@openzeppelin --target solana contracts_flat/OZimport.sol 
> 
> /home/sum/solang-test/contracts_flat/OZimport.sol:848:5-10: error: unrecognised token `using', expected "(", ";", "[", "abstract", "address", "bool", "bytes", "contract", "enum", "event", "false", "function", "import", "interface", "library", "mapping", "payable", "pragma", "string", "struct", "this", "true", Bytes, DocComment, Int, Uint, address, hexnumber, hexstring, identifier, number, rational, string
> Line 848:
> 	using EnumerableSet for EnumerableSet.AddressSet;
> 	^^^^^
> error: not all contracts are valid
> ```

This is on line 848, but the file you quoted contains only 10 or so lines. Can you attach the entire file please, thanks

**Comment by SvenMeyer** [comment]
@seanyoung yes, the error message was from my full contract, but I extracted only the relevant lines for the post.

However, looks like it has been fixed already in Solang version v0.1.9-58-g991ddaf1

```
$ docker run --rm -it -v $PWD:/project   ghcr.io/hyperledger-labs/solang   -o /project/build --importmap @openzeppelin=/project/node_modules/@openzeppelin --target solana -v /project/contracts/OZimport.sol 
info: Solang version v0.1.9-58-g991ddaf1
/project/node_modules/@openzeppelin/contracts/utils/EnumerableSet.sol:3:8-23: debug: pragma ‘solidity’ is ignored
Line 3:
	pragma solidity ^0.7.0;
	-------^^^^^^^^^^^^^^^
/project/node_modules/@openzeppelin/contracts/utils/EnumerableSet.sol:28:4-29:23: debug: found library ‘EnumerableSet’
Line 28:
	*/library EnumerableSet {
	--^^^^^^^^^^^^^^^^^^^^^^
/project/contracts/OZimport.sol:4:1-5:19: debug: found contract ‘OZimport’
Line 4:
	contract OZimport {
	^^^^^^^^^^^^^^^^^^
info: contract OZimport uses at least 16 bytes account data
info: Saving binary /project/build/bundle.so for contracts: OZimport
info: Generating Ethereum ABI for contract OZimport
info: Saving ABI /project/build/OZimport.abi for contract OZimport

```

**Comment by seanyoung** [comment]
Perfect!
